lessons/python_packages/lesson.md

```diff
@@ -0,0 +1,258 @@
+# Creating Packages in Python
+
+- Presenter: John Ladan
+- Sources: Elise Olson, [The Hitchhiker's Guide to Packaging](http://the-hitchhikers-guide-to-packaging.readthedocs.org/en/latest/quickstart.html), [*Bill Mills*](https://github.com/BillMills/pythonPackageLesson)
+- Research Field: any
+- Topic: A quick introduction to creating packages in Python
+
+## Why are we doing this?
+
+One of the most powerful things about coding for the sciences is that it costs
+nothing to re-use code we've written in the past, allowing us to build on
+past work rather than starting over every project or paper.
+
+However, one practice we see again and again is copying and pasting code from one project into another. Sometimes it will just be a function, other times (*cough*MATLAB*cough*) it's files.
+
+Assembling
+code in packages makes it really easy to re-use old code: all the scripts and functions end up in a central location and can be called and imported from anywhere on the computer -
+just like the famous packages `numpy` or `matplotlib`.
+
+## Setting up
+
+In this tutorial, we'll walk through the creation of a simple package from some code that's just lying around. You can view and clone the [demo package](https://github.com/jladan/package_demo). For an unpackaged state, checkout the `unpackaged` tag.
+
+## 1. The basics
+
+The most basic directory structure for a Python package looks like this:
+
+```
+project
+|
+|__ setup.py
+|
+|__ myPackage
+     |
+     |__ somePython.py
+     |__ __init__.py
```

Comment:
Yes, I know; I'm saying they're running together visually. The space between the two underscores is only slightly bigger than the space between the pipe and the underscore, so adding a couple spaces would help make the two more distinct.